pkg/kubelet/kubelet.go

```diff
@@ -1925,9 +1894,15 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		// this conveniently retries any Deferred resize requests
 		// TODO(vinaykul,InPlacePodVerticalScaling): Investigate doing this in HandlePodUpdates + periodic SyncLoop scan
 		//     See: https://github.com/kubernetes/kubernetes/pull/102884#discussion_r663160060
-		pod, err = kl.handlePodResourcesResize(pod, podStatus)
-		if err != nil {
-			return false, err
+		allocatedPods := kl.getAllocatedPods()
+
+		if allocatedPod, resizeNotAllowed := kl.disallowResizeForSwappableContainers(pod); resizeNotAllowed {
```

Comment:
ideally this check would go into `HandlePodResourcesResize` but there are some dependency cycles that need to be reconciled before we can do that

Reply:
that said, this is kind of a moot point / no longer relevant after https://github.com/kubernetes/kubernetes/pull/131612 anyways